### ISSUE TITLE
Progress bar: Option to only ever increase?

### ISSUE THREAD

**Issue by garrett** [patternfly/patternfly-react#4120]
Sometimes progress data that is fed to progress bars is not the highest quality. It sometimes jitters a bit, causing progress bars to "time travel". When this happens, it really drops a user's confidence level in a the task being performed and the product at large.

As every application that uses a progress bar to indicate progress should always show bars moving forward, it might make sense to have an option to enforce forward movement within PatternFly React.

This _wouldn't_ solve cases where the jitters are larger spikes (such as when a progress bar jumps to 90% and hangs out there for a while — there would probably have to be data-specific smoothing to prevent these major jumps). But having an always-move-forward option would definitely help to make sure applications feel more polished.

As progress bars can be used for status indicators as well (disk space, CPU usage, etc.), this would need to be an optional setting.

**Comment by mcarrano** [comment]
I agree this makes sense as an optional setting so that a new value passes to the component must be >= to the current value. Should not require any design or core changes.

cc @LHinson @tlabaj

**Comment by jcaianirh** [comment]
@mcarrano i looked at this, and am wondering if this should be a demo rather than an in component feature? The component as is doesn't really manipulate state or props much which is common for most of the components. I think it might make more sense to show this as a demo. @tlabaj what do you think?
